modules/imgimport/include/csvreading.h

```diff
@@ -0,0 +1,22 @@
+#ifndef CSV_READING_H_INCLUDED
+#define CSV_READING_H_INCLUDED
+
+#include "metadata.h";
+
+struct Metadatalinkedlist{
+    Metadata a;
+    Metadatalinkedlist* next;
+};
+
+/**
+* @brief this is for reading telemetry logs from the plane and store them into a linked list 
+*
+*
+* @param filename the relative path and name of the file
+* @error if the file opening fails, print a fatal log "error: failed to open the cdv file" and return NULL
```

Comment:
There is however an exception command, though you are not actually throwing an exception here.
That being said, an exception might be the better way to handle it since currently there's no way for the program to recognize the difference between a file not being found and finding a file with no log entries.